assets/src/scripts/lib/ajax.js

```diff
@@ -39,3 +39,50 @@ export function get(url, resolveWith='response') {
         req.send();
     });
 }
+
+/**
+ * Executes an HTTP GET request.
+ * @param  {String} url         URL to request
+ * @param  {String} resolveWith XMLHttpRequest attribute to resolve promise
+ *                              with. e.g., "response" or "responseXML"
+ * @return {Promise}
+ */
+export function post(url, sampleData, resolveWith='response', contentType = '') {
+    // Return a new promise.
+    return new Promise(function (resolve, reject) {
+        try {
+            // Do the usual XHR stuff
+            var req = new XMLHttpRequest();
+            req.open('POST', url);
+            if (contentType != '') {
+                req.setRequestHeader('Content-Type', contentType);
+            }
+
+            req.onload = function () {
+                // This is called even on 404 etc
+                // so check the status
+                if (req.status == 200) {
+                    // Resolve the promise with the response text
+                    resolve(req[resolveWith]);
+                } else {
+                    // Otherwise reject with the status text
+                    // which will hopefully be a meaningful error
+                    if (window.ga) {
+                        window.ga('send', 'event', 'serviceFailure', req.status, url);
+                    }
+                    reject(Error(`Failed with status ${req.status}: ${req.statusText}`));
+                }
+            };
+
+            // Handle network errors
+            req.onerror = function () {
+                reject(Error('Network Error'));
+            };
+
+            // Make the request
+            req.send(sampleData);
+        } catch (e) {
+            console.log(e);
```

Comment:
Was this for debugging? It would be good to remove, because it will prevent the promise from resolving.

Reply:
removed.